src/Control/Lens/Setter.hs

```diff
@@ -342,6 +342,10 @@ cloneIndexedSetter l pafb = taintedDot (runIdentity #. l (Indexed $ \i -> Identi
 -- >>> over _1 show (10,20)
 -- ("10",20)
 --
+-- Like 'fmap', @over@ is normally lazy in the result(s) of calling the
```

Comment:
I'm also on the edge with this "warning".

Shouldn't the strict version also have a similar warning that it *actually* forces all the computations up front. I.e. `over' traverse f aList` is most likely a wrong thing to do. (And the examples used are contrived).

Reply:
Yes, they're contrived. Should I give a datatype definition and `makeLenses` and all that jazz? A more realistic case might be a record with a lazy field. If you use `over` several times, you'll build a thunk chain in there. `over' traverse` isn't too far from what, say, `Data.Map.Strict.map` does, though it's probably less efficient.

Reply:
That's a problem. I think in majority of relevant cases you'd rather make the field strict. In fact, there are people which would prefer `Data.Map.Strict` to be a separate datatype. (The type system doesn't help to keep the `Map` thunk free.)

i.e. while `over'` is useful, I don't think it should be commonly used.

Reply:
The only case I can think out of my head is `Seq`. Where `over' traverse` doesn't leave thunks, but AFAIK ruins asymptotics.

Reply:
@phadej `over'` could quite reasonably be used with `ix` for `Array`, `SmallArray` (if that has an instance), `Vector`, `Seq`, `Map`, or `IntMap`. I imagine some folks just use `at` instead, but that's semantically overkill for just forcing.